homeassistant/components/knx/device_trigger.py

```diff
@@ -99,6 +99,8 @@ def async_call_trigger_action(telegram: TelegramDict) -> None:
             {"trigger": {**trigger_data, **telegram}},
         )
 
-    return knx.telegrams.async_listen_telegram(
-        async_call_trigger_action, name="KNX device trigger call"
+    return async_dispatcher_connect(
```

Comment:
Side note: Every device trigger needs to be based off a regular trigger like an event or integration specific trigger. Device triggers are just a device abstraction over a regular trigger and are not allowed to be standalone. I don't see any integration specific triggers in the KNX integration. See eg webostv integration for how to create an integration specific trigger.

Reply:
Yes, Knx doesn't have any integration specific triggers. This was never mentioned in a review, nor could I find anything about that in the developer documentations. I'll try to have a look into in when I have some time left. Are these always looked up in a `<integration>/trigger.py` file?

Until then, I think this bugfix should still be merged as it doesn't change any behaviour on the current device trigger except fixing integration reloads.

Reply:
Yes, the module should be named `trigger.py`.

My comment is a side note for the future.

Reply:
It says here that device automations are built on top of core concepts and are not exposing new functionality.

> Device Automations provide users with a device-centric layer on top of the core concepts of Home Assistant.

> Device automations are not exposing extra functionality but are a way for users to not have to learn new concepts.

https://developers.home-assistant.io/docs/device_automation_index

Reply:
Ah thanks. So I can add a trigger like `knx.telegram`? That would be awesome. 

I initially implemented this to avoid using HA events for the same effect, to not spam the event bus (and db) with unneeded Knx telegram information (~1-5 average, max 50 per second). I guess I can filter the integration specific trigger the same way we are doing for the device trigger, so this should be fine. (And not require any yaml as events would 😬)

Reply:
Triggers are always available to be setup in automations via YAML if the user so wishes. Maybe that's not what you mean?

Reply:
Or right, that's not what I mean. We currently have Events, but users would need to configure address filters in yaml for it to work (to not spam). 
With a trigger I think we could provide a form (/yaml) directly in the automation builder for this